src/Sarif/ZipArchiveArtifact.cs

```diff
@@ -30,6 +30,15 @@ public ZipArchiveArtifact(ZipArchive archive, ZipArchiveEntry entry, ISet<string
 
         public Uri Uri => this.uri;
 
+        public bool IsBinary
+        {
+            get
+            {
+                string extension = Path.GetExtension(Uri.ToString());
+                return this.binaryExtensions.Contains(extension);
+            }
+        }
+
         public Stream Stream
```

Comment:
I don't know why we're splitting this change into an IsBinary implementation based on extension, followed up by a Peekable-backed check in a separate change.  That's not an objection.  I'm just worried I don't see a risk others see.  Is there a reason why ZipArchiveArtifact.IsBinary can't follow the pattern in EnumeratedArtifact, and immediately call GetArtifactData()?

Reply:
ye the final goal should be using the peakable stream for binary sniffing. this is just a quick fix to release the bug fix to unblock for ADO service team. Will create work item for it.

Reply:
Thanks for that.  If you can move it to active so it's obvious it's at the front of my plate, I'd appreciate it.

Reply:
Also, thanks for the clarification on the motivation for the sequencing.   This makes a lot of sense to me as a QFE.  Part of me wondered why I shouldn't be offering to quickly add the Peekable stuff into this change.  It shouldn't be hard, but it would not be as safe of a change as this conservative approach.

Reply:
User Story 2142116: Update `ZipArchiveArtifact` to reuse peakable steam implementation in `EnumeratedArtifact`